### PR TITLE
sql: add kv_catalog_* virtual tables

### PR DESCRIPTION
**Pull request by postamar**
This commit introduces the following virtual tables, which each mirror
more or less exactly a system table containing catalog information:
  - crdb_internal.kv_catalog_descriptor mirrors system.descriptor,
  - crdb_internal.kv_catalog_namespace mirrors system.namespace,
  - crdb_internal.kv_catalog_zones mirrors system.zones,
  - crdb_internal.kv_catalog_comments mirrors system.comments.

Each virtual table is constructed by overlaying the data in the system
table with the in-memory data in the transaction's descs.Collection.
This may include virtual object descriptors and comments as well.

This makes it possible to immediately get rid of the predefined_comments
virtual table and simply replace its usages with kv_catalog_comments.
Longer term, these new virtual tables are expected to become useful to
test declarative schema changer behavior, also they might prove
convenient for troubleshooting schema-related incidents because the
descriptors and zone configs are readily exposed as JSON values.

Informs #88571.

Release note: None

### CODE REVIEW

File: pkg/sql/crdb_internal.go [hunk 1]
```diff
@@ -4895,51 +4900,154 @@ CREATE TABLE crdb_internal.kv_store_status (
...
-		vEntries := vt.getSchemas()
-		vSchemaNames := vt.getSchemaNames()
+		all, err := p.Descriptors().GetAll(ctx, p.Txn())
+		if err != nil {
+			return err
+		}
+		// Delegate privilege check to system table.
+		{
+			sysTable := all.LookupDescriptor(systemschema.DescriptorTable.GetID())
+			if p.CheckAnyPrivilege(ctx, sysTable) != nil {
```
Comment (ajwerner) [inline pkg/sql/crdb_internal.go:4920]: Should this be CheckAnyPrivilege or should it be something else? Namely select

Comment (postamar) [reply pkg/sql/crdb_internal.go:4920]: Oh, good point. I shouldn't have blindly copied that.

**Review by ajwerner — APPROVED** [review]
LGTM

File: pkg/sql/catalog/schemadesc/public_schema_desc.go [hunk 2]
```diff
@@ -42,25 +42,21 @@ type public struct {
...
-	return catpb.NewPublicSchemaPrivilegeDescriptor()
-}
-func (p public) GetRawBytesInStorage() []byte { return nil }
-
-// GetObjectType implements the Object interface.
-func (p public) GetObjectType() privilege.ObjectType {
-	return privilege.Schema
-}
+func (p public) GetID() descpb.ID                     { return keys.PublicSchemaID }
+func (p public) GetParentID() descpb.ID               { return descpb.InvalidID }
```
[the remark this answers is not in the document]
Comment (postamar) [reply pkg/sql/catalog/schemadesc/public_schema_desc.go:46]: No, there's a mildly annoying edge case involving restores, something about the temporary system database also not having a public schema descriptor.

**Comment by postamar** [comment]
Thanks for the review!

bors r+

**Pull request merged**